api/pkg/artifact/artifact.go

```diff
@@ -133,24 +178,120 @@ func (gac *GcsArtifactClient) DeleteArtifact(ctx context.Context, url string) er
 	return nil
 }
 
+type S3ArtifactClient struct {
+	URLScheme
+	Type   string
+	client *s3.Client
+}
+
+func NewS3ArtifactClient() (Service, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("%s,failed loading s3 config for the artifact client", err.Error())
+	}
+	client := s3.NewFromConfig(cfg, func(o *s3.Options) {
+		o.BaseEndpoint = aws.String(os.Getenv("AWS_ENDPOINT_URL"))
+	})
+	return &S3ArtifactClient{
+		Type:      s3ArtifactClientType,
+		URLScheme: s3URLScheme,
+		client:    client,
+	}, nil
+}
+
+func (s3c *S3ArtifactClient) GetType() string {
+	return s3c.Type
+}
+
+func (s3c *S3ArtifactClient) ReadArtifact(ctx context.Context, url string) ([]byte, error) {
+	u, err := s3c.ParseURL(url)
+	if err != nil {
+		return nil, err
+	}
+
+	reader, err := s3c.client.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(u.Bucket),
+		Key:    aws.String(u.Object),
+	})
+	if err != nil {
+		var nsk *types.NoSuchKey
+		if errors.As(err, &nsk) {
+			return nil, ErrObjectNotExist
+		}
+		return nil, err
+	}
+	defer reader.Body.Close() //nolint:errcheck
```

Comment:
is there case that the body is empty or nil?

Reply:
Nope the [body](https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/api_op_GetObject.go#L440) implements the `Closer` interface which has a `Close()` method.